### PR TITLE
[CDAP-6370] - Adds integration of macros for a pipeline from backend with hydrator UI

### PR DESCRIPTION
**Pull request by ajainarayanan**
Bamboo build: http://builds.cask.co/browse/CDAP-DRC4123-1

### CODE REVIEW

**Comment by elfenheart** [comment]
Should we attach the macros to preference instead so that it will persist?

File: cdap-ui/app/features/hydratorplusplus/controllers/detail/top-panel-ctrl.js [hunk 1]
```diff
@@ -53,6 +58,81 @@ class HydratorDetailTopPanelController {
...
+        .map(macro => macro + ':')
+        .join(',');
+    };
+    let {namespace, app} = this.HydratorPlusPlusDetailRunsStore.getParams();
+    let preferenceParams = {
+      namespace: this.$state.params.namespace,
+      appId: this.$state.params.pipelineId,
+      scope: this.$scope
+    };
+    const errorHandler = err => console.log('ERRRO', err);
```
Comment (elfenheart) [inline cdap-ui/app/features/hydratorplusplus/controllers/detail/top-panel-ctrl.js]: typo on ERROR

File: cdap-ui/app/features/hydratorplusplus/controllers/detail/top-panel-ctrl.js [hunk 2]
```diff
@@ -53,6 +58,81 @@ class HydratorDetailTopPanelController {
...
+        errorHandler
+      )
+      .then(
+        res => this.syncPreferencesStoreWithMacros(res),
+        errorHandler
+      );
+  }
+  syncPreferencesStoreWithMacros(appPreferences = {}) {
+    try {
+      appPreferences = JSON.parse(angular.toJson(appPreferences));
```
Comment (elfenheart) [inline cdap-ui/app/features/hydratorplusplus/controllers/detail/top-panel-ctrl.js:112]: why is this necessary? seems like we are turning something to JSON and parsing it back?

File: cdap-ui/app/features/hydratorplusplus/controllers/detail/top-panel-ctrl.js [hunk 3]
```diff
@@ -201,11 +297,22 @@ class HydratorDetailTopPanelController {
...
-          this.myAlertOnValium.show({
-            type: 'danger',
-            title: 'Unable to schedule the pipeline',
-            content: angular.isObject(err) ? err.data: err
-          });
+          if (!this.macroError) {
+            this.myAlertOnValium.show({
+              type: 'danger',
+              title: 'Unable to schedule the pipeline',
+              content: angular.isObject(err) ? err.data: err
```
Comment (elfenheart) [inline cdap-ui/app/features/hydratorplusplus/controllers/detail/top-panel-ctrl.js]: nit: space before `:`

**Comment by elfenheart** [comment]
`app/features/hydratorplusplus/controllers/detail/top-panel-ctrl.js: line 270, col 9, 'prom' is defined but never used.`

File: cdap-ui/app/features/hydratorplusplus/controllers/detail/top-panel-ctrl.js (hunk 2 above)
Comment (ajainarayanan) [inline cdap-ui/app/features/hydratorplusplus/controllers/detail/top-panel-ctrl.js:112]: So `appPreferences` is returned from `$resource` and it has `$$hashKey` and similar angular specific variables. `angular.toJson` removes them and gives back a stringified json which we are parsing again to be a JSON

**Comment by elfenheart** [comment]
👍  LGTM

**Comment by ajainarayanan** [comment]
Rebased it against release/3.5 and demoed it to @nitinmotgi . Hence merging.
